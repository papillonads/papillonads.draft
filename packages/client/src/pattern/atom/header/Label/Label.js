import React from 'react'
import { string } from 'prop-types'
import cn from 'classnames'
import styles from './Label.scss'

const Label = ({ className }) => {
  const labelClassList = cn(className, styles.label)

  return <div className={labelClassList}>Label</div>
}

Label.defaultProps = {
  className: '',
}

Label.propTypes = {
  className: string,
}.isRequired

export default Label
