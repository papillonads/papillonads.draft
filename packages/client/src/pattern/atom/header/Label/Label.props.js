import { string } from 'prop-types'

export const labelDefaultProps = {
  className: '',
}

export const labelPropTypes = {
  className: string,
  text: string,
}
