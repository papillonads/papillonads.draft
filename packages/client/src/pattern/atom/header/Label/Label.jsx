import React from 'react'
import { labelDefaultProps, labelPropTypes } from './Label.props'

export const Label = ({ className, text }) => <div className={className}>{text}</div>

Label.defaultProps = labelDefaultProps

Label.propTypes = labelPropTypes
