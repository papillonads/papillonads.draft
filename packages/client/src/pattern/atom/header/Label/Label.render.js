import React from 'react'
import Label from './Label'

const defaultRender = <Label text="some-text" />

export { defaultRender }
