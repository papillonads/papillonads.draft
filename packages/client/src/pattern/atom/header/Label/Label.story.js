import { storiesOf } from '@storybook/react'
import { defaultRender } from './Label.render'
import { getInfo } from '../../../../../../../.storybook/library'

const defaultRenderInfo = getInfo('default')

storiesOf('Patttern/Atom/Header/Label', module).add(defaultRenderInfo.text, () => defaultRender, defaultRenderInfo.parameters)
