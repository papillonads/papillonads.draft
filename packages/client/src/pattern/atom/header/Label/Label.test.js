import { defaultRender } from './Label.render'

describe('<Label />', () => {
  describe('Snaphot', () => {
    test('must match defaultRender', () => {
      expect(global.renderToJSON(defaultRender)).toMatchSnapshot()
    })
  })
})
