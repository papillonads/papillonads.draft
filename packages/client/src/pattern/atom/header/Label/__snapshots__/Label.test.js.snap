// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Snaphot must match defaultRender 1`] = `null`;
